test/Microsoft.AspNetCore.Mvc.DataAnnotations.Test/Internal/DataAnnotationsMetadataProviderTest.cs

```diff
@@ -167,6 +168,64 @@ public void CreateBindingMetadata_EditableAttributeTrue_SetsReadOnlyFalse()
         }
 
         [Fact]
+        public void CreateDisplayMetadata_DisplayNameAttribute_UsesDisplayName()
```

Comment:
Do this simple check w/ an addition to `DisplayDetailsData` instead. Save the more involved tests for the `IStringLocalizer` (or `ResourceType` where applicable) complications.